### PR TITLE
prairielearn module test coverage; nan/inf edge case; matlab format

### PR DESCRIPTION
**Pull request by reteps**
🥳 improve our test coverage

### CODE REVIEW

File: apps/prairielearn/python/prairielearn/core.py [hunk 1]
```diff
@@ -1773,6 +1782,10 @@ def is_correct_scalar_dd(a_sub: ArrayLike, a_tru: ArrayLike, digits: int = 2) ->
         imag_comp = is_correct_scalar_dd(a_sub.imag, a_tru.imag, digits=digits)  # type: ignore
         return real_comp and imag_comp
 
+    if np.abs(a_tru) == math.inf:
```
Comment (reteps) [inline apps/prairielearn/python/prairielearn/core.py:1785]: we were missing handling for infinity and nan.

File: apps/prairielearn/python/prairielearn/core.py [hunk 2]
```diff
@@ -1097,6 +1099,11 @@ def numpy_to_matlab_sf(
 
         - a number (float or complex)
         - a 2D ndarray (float or complex)
+
+    The style argument must be one of three values:
+    - legacy: formats 1d arrays with commas and 2d arrays with spaces
```
Comment (reteps) [inline apps/prairielearn/python/prairielearn/core.py:1104]: This behavior was unclear, so I made some more sane parameters to choose.

Comment (reteps) [reply apps/prairielearn/python/prairielearn/core.py:1104]: MATLAB accepts all of these, I checked with an online compiler.

File: apps/prairielearn/python/test/core_test.py [hunk 3]
```diff
@@ -636,96 +677,574 @@ def test_string_to_2darray_invalid(
...
+    assert pl.is_correct_ndarray2d_sf(submitted, true, digits) == expected
+
+
+def test_load_extension() -> None:
+    """Test loading extensions with the load_extension function."""
+    director = Path(__file__).parent
+    controller = "dummy_extension.py"
+
+    # Create mock data with extension info
+    data: pl.QuestionData = {
```
Comment (eliotwrobson) [inline apps/prairielearn/python/test/core_test.py:1099]: I think you can use the question data currently given in the test fixture and add the relevant entries instead of creating a new one from scratch.

File: apps/prairielearn/python/test/core_test.py [hunk 4]
```diff
@@ -527,12 +539,42 @@ def test_string_from_number_sigfig(
...
+def test_numpy_to_matlab_sf(
+    value: Any, args: dict[str, Any], expected_output: str
+) -> None:
     assert pl.numpy_to_matlab_sf(value, **args) == expected_output
 
 
+@pytest.mark.parametrize(
+    ("value", "args", "expected_output"),
+    [
+        (0.0123, {}, "0.01"),
```
Comment (eliotwrobson) [inline apps/prairielearn/python/test/core_test.py:569]: Why only one test case here?

**Review by eliotwrobson — APPROVED** [review]
A couple small items but changes seem low impact and tests look reasonable 👍

Looks like coderabbit has some suggestions, might be worth adding.

File: apps/prairielearn/python/test/core_test.py (hunk 4 above)
Comment (reteps) [reply apps/prairielearn/python/test/core_test.py:569]: This was the only part of coverage not hit.

File: apps/prairielearn/python/test/core_test.py (hunk 3 above)
Comment (reteps) [reply apps/prairielearn/python/test/core_test.py:1099]: Where is this defined so that the other test cases work? I'm very confused there

Comment (eliotwrobson) [reply apps/prairielearn/python/test/core_test.py:1099]: https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/python/test/conftest.py
